test/extensions/filters/udp/dns_filter/dns_filter_fuzz_test.cc

```diff
@@ -0,0 +1,68 @@
+#include "common/common/logger.h"
+
+#include "extensions/filters/udp/dns_filter/dns_filter.h"
+
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+namespace {
+
+const std::string generateQuery(FuzzedDataProvider* data_provider) {
+  size_t query_size = data_provider->ConsumeIntegralInRange<size_t>(0, 512);
+  return data_provider->ConsumeRandomLengthString(query_size);
+}
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  FuzzedDataProvider data_provider(buf, len);
+  const bool recurse = data_provider.ConsumeIntegralInRange<bool>(0, 1);
```

Comment:
a `ConsumeBool` also exists, but it's the same

Reply:
Yes, I should have scrolled *up* in one of the links you sent.  I will update this.